substrate-archive/src/archive.rs

```diff
@@ -73,116 +128,192 @@ where
 }
 
 pub struct ArchiveBuilder<B, R, D, DB> {
 	_marker: PhantomData<(B, R, D, DB)>,
-	/// Chain spec describing the chain
-	chain_spec: Option<Box<dyn ChainSpec>>,
-	/// Path to the rocksdb database
-	chain_data_path: Option<PathBuf>,
-	/// url to the Postgres Database
-	pg_url: Option<String>,
-	/// how much Cache should Rocksdb Keep
-	cache_size: usize,
-	/// number of threads to spawn for block execution
-	block_workers: usize,
-	/// Number of 64KB Heap pages to allocate for wasm execution
-	wasm_pages: u64,
-	/// maximum amount of blocks to index at once
-	pub max_block_load: u32,
-	/// Enable state tracing while also specifying the targets
-	/// and directory where the WASM runtimes are stored.
-	pub wasm_tracing: Option<TracingConfig>,
+	config: ArchiveConfig,
 }
 
 impl<B, R, D, DB> Default for ArchiveBuilder<B, R, D, DB> {
 	fn default() -> Self {
-		let num_cpus = num_cpus::get();
-		Self {
-			_marker: PhantomData,
-			chain_spec: None,
-			chain_data_path: None,
-			pg_url: None,
-			cache_size: 128,                  // 128 MB
-			block_workers: num_cpus,          // the number of logical cpus in the system
-			wasm_pages: 64 * num_cpus as u64, // 64 * (number of logic cpu's)
-			max_block_load: 100_000,          // 100_000 blocks to index at once
-			wasm_tracing: None,
-		}
+		Self { _marker: PhantomData, config: ArchiveConfig::default() }
 	}
 }
 
 impl<B, R, D, DB> ArchiveBuilder<B, R, D, DB> {
-	/// Specify a chain spec for storing metadata about the running archiver
-	/// in a persistant directory.
+	/// Creates a archive builder with the given config.
+	pub fn with_config(config: Option<ArchiveConfig>) -> Self {
+		if let Some(config) = config {
+			Self { _marker: PhantomData, config }
+		} else {
+			Self::default()
+		}
+	}
+
+	/// Specify a chain spec name and id for storing metadata about the running archiver
+	/// in a persistent directory.
 	///
 	/// # Default
-	/// Defaults to storing metadata in a temporary directory.
+	/// defaults to storing metadata in a temporary directory.
 	pub fn chain_spec(mut self, spec: Box<dyn ChainSpec>) -> Self {
-		self.chain_spec = Some(spec);
+		self.config.chain.spec = Some(ChainSpecConfig { name: spec.name().into(), id: spec.id().into() });
 		self
 	}
 
 	/// Set the chain data backend path to use for this instance.
 	///
 	/// # Default
 	/// defaults to the environment variable CHAIN_DATA_DB
-	pub fn chain_data_path<S: Into<PathBuf>>(mut self, path: Option<S>) -> Self {
-		self.chain_data_path = path.map(Into::into);
+	pub fn chain_data_path<S: Into<PathBuf>>(mut self, path: S) -> Self {
+		self.config.chain.data_path = Some(path.into());
+		self
+	}
+
+	/// Set the amount of cache RocksDB should keep.
+	///
+	/// # Default
+	/// defaults to 128MB
+	pub fn cache_size(mut self, cache_size: usize) -> Self {
+		self.config.chain.cache_size = cache_size;
+		self
+	}
+
+	/// Set the path of RocksDB secondary directory according to the chain spec.
```

Comment:
```suggestion
	/// Set the path to the secondary RocksDB database directory according to the chain spec.
```

"according to the chain spec": not sure what is meant by this. Relative to the chain spec?

Reply:
if you specify the path `./substrate-archive/rocksdb_secondary` and chain spec,
the actual path will be `./substrate-archive/rocksdb_secondary/<chain-spec-name>/<chain-spec-id>`.
I will add this comment.